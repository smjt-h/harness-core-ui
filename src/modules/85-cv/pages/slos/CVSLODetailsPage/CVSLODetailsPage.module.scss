/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.tabContainer {
  :global {
    .bp3-tab-list {
      background-color: var(--white);
      padding: var(--spacing-none) var(--spacing-xlarge);
      align-items: center;
      border-bottom: 1px solid var(--grey-200);
    }

    .bp3-tab-panel {
      margin-top: var(--spacing-none) !important;
    }
  }

  .pageBody {
    --page-header-height: 140px !important;
    height: calc(100vh - 140px) !important;
    max-height: calc(100vh - 140px) !important;
  }
}
