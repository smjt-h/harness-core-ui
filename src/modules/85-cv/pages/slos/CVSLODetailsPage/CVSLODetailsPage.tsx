/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useRef } from 'react'
import { useParams, useHistory } from 'react-router-dom'
import { Container, FlexExpander, Page, Tabs } from '@harness/uicore'
import { useStrings } from 'framework/strings'
import { useDeleteSLOData, useGetSLODetails, useResetErrorBudget } from 'services/cv'
import routes from '@common/RouteDefinitions'
import { useQueryParams } from '@common/hooks'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import { getErrorMessage, getSearchString } from '@cv/utils/CommonUtils'
import CVCreateSLO from '@cv/pages/slos/components/CVCreateSLO/CVCreateSLO'
import HeaderTitle from './views/HeaderTitle'
import DetailsPanel from './DetailsPanel/DetailsPanel'
import TabToolbar from './DetailsPanel/views/TabToolbar'
import { SLODetailsPageTabIds } from './CVSLODetailsPage.types'
import css from './CVSLODetailsPage.module.scss'

const CVSLODetailsPage: React.FC = () => {
  const history = useHistory()
  const { getString } = useStrings()
  const { accountId, orgIdentifier, projectIdentifier, identifier } = useParams<
    ProjectPathProps & { identifier: string }
  >()
  const { tab = SLODetailsPageTabIds.Details, monitoredServiceIdentifier } =
    useQueryParams<{ tab?: SLODetailsPageTabIds; monitoredServiceIdentifier?: string }>()

  const projectIdentifierRef = useRef<string>()
  useEffect(() => {
    if (projectIdentifierRef.current && projectIdentifierRef.current !== projectIdentifier) {
      history.push(routes.toCVSLOs({ accountId, orgIdentifier, projectIdentifier }))
      return
    }

    projectIdentifierRef.current = projectIdentifier
  }, [accountId, orgIdentifier, projectIdentifier, history])

  const {
    data,
    loading: sloDetailsLoading,
    error,
    refetch
  } = useGetSLODetails({
    identifier,
    queryParams: {
      accountId,
      orgIdentifier,
      projectIdentifier
    }
  })

  const { mutate: resetErrorBudget, loading: resetErrorBudgetLoading } = useResetErrorBudget({
    identifier: '',
    queryParams: {
      accountId,
      orgIdentifier,
      projectIdentifier
    }
  })

  const { mutate: deleteSLO, loading: deleteSLOLoading } = useDeleteSLOData({
    queryParams: {
      accountId,
      orgIdentifier,
      projectIdentifier
    }
  })

  const onTabChange = (nextTab: SLODetailsPageTabIds): void => {
    /* istanbul ignore else */ if (nextTab !== tab) {
      history.push({
        pathname: routes.toCVSLODetailsPage({
          identifier,
          accountId,
          orgIdentifier,
          projectIdentifier
        }),
        search: getSearchString({ tab: nextTab, monitoredServiceIdentifier })
      })
    }
  }

  const { description, sloDashboardWidget, timeRangeFilters } = data?.data ?? {}
  const loading = sloDetailsLoading || resetErrorBudgetLoading || deleteSLOLoading

  const breadcrumbLinks = [
    {
      url: routes.toCVSLOs({ accountId, orgIdentifier, projectIdentifier }),
      label: getString('cv.SLO')
    }
  ]

  return (
    <>
      <Page.Header
        size="large"
        title={<HeaderTitle loading={sloDetailsLoading} title={sloDashboardWidget?.title} description={description} />}
        breadcrumbs={<NGBreadcrumbs links={breadcrumbLinks} />}
      />
      <Container className={css.tabContainer}>
        <Tabs
          id="slo-details-page-tabs"
          selectedTabId={tab}
          onChange={onTabChange}
          tabList={[
            {
              id: SLODetailsPageTabIds.Details,
              title: getString('details'),
              panel: (
                <DetailsPanel
                  loading={loading}
                  errorMessage={getErrorMessage(error)}
                  retryOnError={() => refetch()}
                  sloDashboardWidget={sloDashboardWidget}
                  timeRangeFilters={timeRangeFilters}
                />
              )
            },
            {
              id: SLODetailsPageTabIds.Configurations,
              title: getString('cv.monitoredServices.monitoredServiceTabs.configurations'),
              panel: (
                <Page.Body
                  loading={loading}
                  error={getErrorMessage(error)}
                  retryOnError={() => refetch()}
                  noData={{
                    when: () => !sloDashboardWidget
                  }}
                  className={css.pageBody}
                >
                  <CVCreateSLO />
                </Page.Body>
              )
            }
          ]}
        >
          <FlexExpander />
          {tab === SLODetailsPageTabIds.Details && sloDashboardWidget && (
            <TabToolbar
              sloDashboardWidget={sloDashboardWidget}
              resetErrorBudget={resetErrorBudget}
              deleteSLO={deleteSLO}
              refetchSLODetails={refetch}
              onTabChange={onTabChange}
            />
          )}
        </Tabs>
      </Container>
    </>
  )
}

export default CVSLODetailsPage
