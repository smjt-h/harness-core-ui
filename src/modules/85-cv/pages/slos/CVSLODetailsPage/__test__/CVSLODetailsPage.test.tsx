/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, RenderResult, screen, waitFor } from '@testing-library/react'
import * as cvServices from 'services/cv'
import { TestWrapper } from '@common/utils/testUtils'
import CVSLODetailsPage from '../CVSLODetailsPage'
import {
  responseSLODashboardDetail,
  testWrapperProps,
  errorMessage,
  pathParams,
  responseSLODashboardDetail2
} from './CVSLODetailsPage.mock'

jest.mock('@cv/pages/slos/components/CVCreateSLO/CVCreateSLO', () => ({
  __esModule: true,
  default: function CVCreateSLO() {
    return <div>MOCKED - CVCreateSLO</div>
  }
}))

jest.mock('@cv/pages/slos/SLOCard/ErrorBudgetGauge', () => ({
  __esModule: true,
  default: function ErrorBudgetGauge() {
    return <span data-testid="error-budget-gauge" />
  }
}))

jest.mock('@cv/pages/slos/SLOCard/SLOCardContent', () => ({
  __esModule: true,
  default: function ErrorBudgetGauge() {
    return <span data-testid="slo-card-content" />
  }
}))

jest.mock(
  '@cv/pages/monitored-service/components/ServiceHealth/components/ChangesAndServiceDependency/components/ChangesTable/ChangesTable',
  () => ({
    __esModule: true,
    default: function ErrorBudgetGauge() {
      return <span data-testid="changes-table" />
    }
  })
)

jest.mock(
  '@cv/pages/monitored-service/components/ServiceHealth/components/ChangesSourceCard/ChangesSourceCard',
  () => ({
    __esModule: true,
    default: function ChangesSourceCard() {
      return <span data-testid="changes-source-card" />
    }
  })
)

const renderComponent = (): RenderResult => {
  return render(
    <TestWrapper {...testWrapperProps}>
      <CVSLODetailsPage />
    </TestWrapper>
  )
}

describe('Test cases for CVSLODetailsPage', () => {
  test('it should render the component and take a snapshot', () => {
    jest
      .spyOn(cvServices, 'useGetSLODetails')
      .mockReturnValue({ data: responseSLODashboardDetail, loading: false } as any)

    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })

  test('it should handle the loading state', () => {
    jest.spyOn(cvServices, 'useGetSLODetails').mockReturnValue({ data: null, loading: true } as any)

    const { container } = renderComponent()

    expect(container.getElementsByClassName('bp3-skeleton')).toHaveLength(4)
  })

  test('is should handle retryOnError for both tabs', async () => {
    const refetch = jest.fn()

    jest
      .spyOn(cvServices, 'useGetSLODetails')
      .mockReturnValue({ data: null, loading: false, error: { message: errorMessage }, refetch } as any)

    renderComponent()

    expect(screen.getByText('details')).toHaveAttribute('aria-selected', 'true')
    expect(screen.getByText(errorMessage)).toBeInTheDocument()

    userEvent.click(screen.getByText('Retry'))

    const { identifier, accountId, orgIdentifier, projectIdentifier } = pathParams

    await waitFor(() => {
      expect(cvServices.useGetSLODetails).toHaveBeenLastCalledWith({
        identifier,
        queryParams: {
          accountId,
          orgIdentifier,
          projectIdentifier
        }
      })
    })

    userEvent.click(screen.getByText('cv.monitoredServices.monitoredServiceTabs.configurations'))

    expect(screen.getByText('cv.monitoredServices.monitoredServiceTabs.configurations')).toHaveAttribute(
      'aria-selected',
      'true'
    )

    userEvent.click(screen.getByText('Retry'))

    await waitFor(() => {
      expect(cvServices.useGetSLODetails).toHaveBeenLastCalledWith({
        identifier,
        queryParams: {
          accountId,
          orgIdentifier,
          projectIdentifier
        }
      })
    })
  })

  test('it should handle the suffix day/days', () => {
    jest
      .spyOn(cvServices, 'useGetSLODetails')
      .mockReturnValue({ data: responseSLODashboardDetail2, loading: false } as any)

    renderComponent()

    expect(screen.getByText('cv.slos.slis.type.availability')).toBeInTheDocument()
  })
})
