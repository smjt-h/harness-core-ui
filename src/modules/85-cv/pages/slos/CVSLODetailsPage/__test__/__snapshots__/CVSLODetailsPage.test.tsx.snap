// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for CVSLODetailsPage it should render the component and take a snapshot 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--large StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <div
        class="PageHeader--breadcrumbsDiv"
      >
        <div
          class="Layout--horizontal StyledProps--main Breadcrumbs--breadcrumbs StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
        >
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/ACCOUNT_ID/cv/orgs/ORG_IDENTIFIER/projects/PROJECT_IDENTIFIER"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                data-icon="cv-main"
              >
                <svg
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M25.933 24.357A12.972 12.972 0 0116 28.969a12.972 12.972 0 01-9.933-4.612l6.025-3.834A5.976 5.976 0 0016 21.969a5.977 5.977 0 003.907-1.446l6.026 3.834zm1.066-1.455l-5.906-3.759a6.001 6.001 0 00-4.193-9.107V3C23.66 3.462 29 9.092 29 15.97a12.94 12.94 0 01-2 6.932zM15.1 3C8.34 3.462 3 9.092 3 15.97c0 2.548.733 4.925 2 6.932l5.907-3.759a6.001 6.001 0 014.192-9.107V3z"
                    fill="url(#cv-main_svg__paint0_linear_20630_159361)"
                    fill-rule="evenodd"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M19.558 13.286a.9.9 0 01-.044 1.272l-4.286 4a.9.9 0 01-1.228 0l-1.714-1.6a.9.9 0 011.228-1.316l1.1 1.027 3.672-3.427a.9.9 0 011.272.044z"
                    fill="#AE82FC"
                    fill-rule="evenodd"
                  />
                  <defs>
                    <lineargradient
                      gradientUnits="userSpaceOnUse"
                      id="cv-main_svg__paint0_linear_20630_159361"
                      x1="7.5"
                      x2="20.926"
                      y1="3"
                      y2="28.987"
                    >
                      <stop
                        stop-color="#BEA1FF"
                      />
                      <stop
                        offset="1"
                        stop-color="#8A36FF"
                      />
                    </lineargradient>
                  </defs>
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                PROJECT_IDENTIFIER
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
          <div
            class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-justifyContent-flex-start"
          >
            <a
              class="Breadcrumbs--breadcrumb"
              href="/account/ACCOUNT_ID/cv/orgs/ORG_IDENTIFIER/projects/PROJECT_IDENTIFIER/slos"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--intent-primary StyledProps--font-size-small"
              >
                cv.SLO
              </p>
            </a>
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey500 StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
              data-icon="main-chevron-right"
            >
              <svg
                height="8"
                viewBox="0 0 8 14"
                width="8"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                  fill="#4d5969"
                />
              </svg>
            </span>
          </div>
        </div>
      </div>
      <div
        class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-evenly StyledProps--flex-alignItems-flex-start"
        style="height: 45px;"
      >
        <h3
          class="StyledProps--font StyledProps--font-h3 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-h5"
          level="3"
        >
          cv.SLOWithName
        </h3>
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey600 StyledProps--font-variation-body2"
        >
          99.9 percent of logins should have &lt;= 50ms latency
        </p>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main tabContainer"
  >
    <div
      class="bp3-tabs Tabs--main Tabs--horizontal"
    >
      <div
        class="bp3-tab-list"
        role="tablist"
      >
        <div
          class="bp3-tab-indicator-wrapper"
          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
        >
          <div
            class="bp3-tab-indicator"
          />
        </div>
        <div
          aria-controls="bp3-tab-panel_slo-details-page-tabs_Details"
          aria-disabled="false"
          aria-expanded="true"
          aria-selected="true"
          class="bp3-tab"
          data-tab-id="Details"
          id="bp3-tab-title_slo-details-page-tabs_Details"
          role="tab"
          tabindex="0"
        >
          details
        </div>
        <div
          aria-controls="bp3-tab-panel_slo-details-page-tabs_Configurations"
          aria-disabled="false"
          aria-expanded="false"
          aria-selected="false"
          class="bp3-tab"
          data-tab-id="Configurations"
          id="bp3-tab-title_slo-details-page-tabs_Configurations"
          role="tab"
          tabindex="0"
        >
          cv.monitoredServices.monitoredServiceTabs.configurations
        </div>
        <span
          style="flex-grow: 1;"
        />
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main tabLink Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey700"
            data-icon="Edit"
          >
            <svg
              height="14"
              viewBox="0 0 16 16"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                fill="currentColor"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                fill="currentColor"
                fill-rule="evenodd"
                opacity="0.3"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            edit
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main tabLink Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-reset StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey700"
            icon="reset"
          >
            <svg
              data-icon="reset"
              height="12"
              viewBox="0 0 16 16"
              width="12"
            >
              <desc>
                reset
              </desc>
              <path
                d="M6 5c0 .55-.45 1-1 1H1c-.55 0-1-.45-1-1V1c0-.55.45-1 1-1s1 .45 1 1v1.74A7.95 7.95 0 018 0c4.42 0 8 3.58 8 8 0 4.06-3.02 7.4-6.94 7.92-.02 0-.04.01-.06.01-.33.04-.66.07-1 .07-4.42 0-8-3.58-8-8 0-.55.45-1 1-1s1 .45 1 1c0 3.31 2.69 6 6 6 .71 0 1.37-.15 2-.38v.01c2.33-.82 4-3.02 4-5.63 0-3.31-2.69-6-6-6-1.77 0-3.36.78-4.46 2H5c.55 0 1 .45 1 1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            cv.resetErrorBudget
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main tabLink Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
            data-icon="audit-trail"
          >
            <svg
              fill="none"
              height="17"
              viewBox="0 0 68 68"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.167 8.5h31.166l8.5 11.333V59.5H14.167v-51z"
                stroke="#383946"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2.833"
              />
              <g
                opacity="0.3"
              >
                <mask
                  fill="#fff"
                  id="audit-trail_svg__a"
                >
                  <path
                    clip-rule="evenodd"
                    d="M25.5 25.5a1.417 1.417 0 000 2.833h17a1.417 1.417 0 100-2.833h-17zm-1.417 9.917c0-.783.635-1.417 1.417-1.417h17a1.417 1.417 0 010 2.833h-17a1.417 1.417 0 01-1.417-1.416zm0 8.5c0-.783.635-1.417 1.417-1.417h17a1.417 1.417 0 010 2.833h-17a1.417 1.417 0 01-1.417-1.416z"
                    fill-rule="evenodd"
                  />
                </mask>
                <path
                  d="M26.917 26.917c0 .782-.635 1.416-1.417 1.416v-5.666a4.25 4.25 0 00-4.25 4.25h5.667zM25.5 25.5c.782 0 1.417.634 1.417 1.417H21.25a4.25 4.25 0 004.25 4.25V25.5zm17 0h-17v5.667h17V25.5zm-1.417 1.417c0-.783.635-1.417 1.417-1.417v5.667a4.25 4.25 0 004.25-4.25h-5.667zm1.417 1.416a1.417 1.417 0 01-1.417-1.416h5.667a4.25 4.25 0 00-4.25-4.25v5.666zm-17 0h17v-5.666h-17v5.666zm0 2.834a4.25 4.25 0 00-4.25 4.25h5.667c0 .782-.635 1.416-1.417 1.416v-5.666zm17 0h-17v5.666h17v-5.666zm4.25 4.25a4.25 4.25 0 00-4.25-4.25v5.666a1.417 1.417 0 01-1.417-1.416h5.667zm-4.25 4.25a4.25 4.25 0 004.25-4.25h-5.667c0-.783.635-1.417 1.417-1.417v5.667zm-17 0h17V34h-17v5.667zm-4.25-4.25a4.25 4.25 0 004.25 4.25V34c.782 0 1.417.634 1.417 1.417H21.25zm4.25 4.25a4.25 4.25 0 00-4.25 4.25h5.667c0 .782-.635 1.416-1.417 1.416v-5.666zm17 0h-17v5.666h17v-5.666zm4.25 4.25a4.25 4.25 0 00-4.25-4.25v5.666a1.417 1.417 0 01-1.417-1.416h5.667zm-4.25 4.25a4.25 4.25 0 004.25-4.25h-5.667c0-.783.635-1.417 1.417-1.417v5.667zm-17 0h17V42.5h-17v5.667zm-4.25-4.25a4.25 4.25 0 004.25 4.25V42.5c.782 0 1.417.634 1.417 1.417H21.25z"
                  fill="currentColor"
                  mask="url(#audit-trail_svg__a)"
                />
              </g>
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            cv.executionLogs
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main tabLink Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey700"
            data-icon="api-docs"
          >
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M4.5 3.415a.5.5 0 01.5-.5h11a.5.5 0 01.4.2l3 4a.5.5 0 01.1.3v14a.5.5 0 01-.5.5h-2.13a.5.5 0 010-1h1.63V7.58l-2.75-3.666H5.5v10.532a.5.5 0 01-1 0V3.415z"
                fill="#fff"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M8.5 9.915a.5.5 0 01.5-.5h6a.5.5 0 010 1H9a.5.5 0 01-.5-.5zm0 3a.5.5 0 01.5-.5h6a.5.5 0 010 1H9a.5.5 0 01-.5-.5z"
                fill="#fff"
                fill-rule="evenodd"
                opacity="0.5"
              />
              <path
                d="M5.006 21.587h-1.15l1.493-4.424h1.423l1.493 4.424h-1.15l-1.038-3.31h-.035l-1.036 3.31zm-.154-1.741h2.402v.812H4.852v-.813zm4.218 1.741v-4.424h1.827c.331 0 .617.064.858.194.242.128.428.308.56.538.13.229.196.495.196.8 0 .305-.067.572-.201.8a1.347 1.347 0 01-.568.53 1.913 1.913 0 01-.877.188H9.737v-.843h.93a.87.87 0 00.403-.084.578.578 0 00.248-.235.728.728 0 00.085-.357.708.708 0 00-.085-.354.566.566 0 00-.248-.233.896.896 0 00-.404-.082h-.527v3.562h-1.07zm5.365-4.424v4.424h-1.069v-4.424h1.07z"
                fill="#fff"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            cv.externalAPICalls
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main tabLink Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-trash StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey700"
            icon="trash"
          >
            <svg
              data-icon="trash"
              height="13"
              viewBox="0 0 16 16"
              width="13"
            >
              <desc>
                trash
              </desc>
              <path
                d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            delete
          </span>
        </button>
      </div>
      <div
        aria-hidden="false"
        aria-labelledby="bp3-tab-title_slo-details-page-tabs_Details"
        class="bp3-tab-panel"
        id="bp3-tab-panel_slo-details-page-tabs_Details"
        role="tabpanel"
      >
        <div
          class="StyledProps--font StyledProps--main PageBody--pageBody pageBody"
          data-name="page-body"
        >
          <div
            class="StyledProps--font StyledProps--main StyledProps--padding-xlarge"
          >
            <div
              class="bp3-card bp3-elevation-0 Card--card serviceDetailsCard"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-card-title StyledProps--color StyledProps--color-grey800 StyledProps--padding-bottom-medium"
              >
                ce.co.autoStoppingRule.review.serviceDetails
              </p>
              <div
                class="Layout--horizontal Layout--layout-spacing-xlarge StyledProps--main"
              >
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey400"
                  >
                    connectors.cdng.monitoredService.label
                  </p>
                  <a
                    href="/account/ACCOUNT_ID/cv/orgs/ORG_IDENTIFIER/projects/PROJECT_IDENTIFIER/monitoringservices/edit/Service_1_Environment_1"
                  >
                    <p
                      class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-primary7"
                    >
                      Service 1
                      <span
                        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--padding-left-xsmall"
                      >
                        /
                        Environment 1
                      </span>
                    </p>
                  </a>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey400"
                  >
                    cv.slos.sliType
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-grey800"
                  >
                    cv.slos.slis.type.latency
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey400"
                  >
                    cv.healthScore
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-grey800"
                  >
                    AppDynamics
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey400"
                  >
                    cv.slos.sloTargetAndBudget.periodType
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-grey800"
                  >
                    cv.slos.sloTargetAndBudget.periodTypeOptions.calendar
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main"
                  style="flex-grow: 1;"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey400"
                  >
                    cv.periodLength
                  </p>
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-small-bold StyledProps--color StyledProps--color-grey800"
                  >
                    7
                     
                    cv.days
                     
                    <span
                      class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey600 StyledProps--font-variation-tiny-semi"
                    >
                      (
                      Apr 2, 2022 12:00 AM
                       - 
                      Apr 9, 2022 12:00 AM
                      )
                    </span>
                  </p>
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--padding-small StyledProps--background-grey100"
                  style="border-radius: 8px; width: 140px;"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey600"
                  >
                    cv.errorBudgetRemaining
                  </p>
                  <h2
                    class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-h4"
                    level="2"
                  >
                    -1112.87
                    %
                  </h2>
                </div>
                <div
                  class="StyledProps--font StyledProps--main StyledProps--padding-small StyledProps--background-grey100"
                  style="border-radius: 8px; width: 140px;"
                >
                  <p
                    class="StyledProps--font StyledProps--main StyledProps--font-variation-tiny-semi StyledProps--color StyledProps--color-grey600"
                  >
                    cv.timeRemaining
                  </p>
                  <h2
                    class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--inline StyledProps--color StyledProps--color-grey800 StyledProps--font-variation-h4"
                    level="2"
                  >
                    2
                  </h2>
                   
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-form-help"
                  >
                    cv.days
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
